Documentation/components/drivers/character/watchdog.rst

```diff
@@ -153,4 +153,51 @@ This command registers an user callback that will be triggered on timeout. It re
 
 .. c:macro:: WDIOC_KEEPALIVE
 
- This command resets the watchdog timer ("ping",  "pet the dog",  "feed the dog"). 
\ No newline at end of file
+ This command resets the watchdog timer ("ping",  "pet the dog",  "feed the dog"). 
+
+Using the Auto-monitor feature to keep the chip alive
```

Comment:
I am confused by this. Is this named correctly? A HW watchdog needs to be "reset"  if not when it times out it is "triggered".  

So what is  Auto-monitor? Is the generating the keep-alive? Then it is not a monitor,

Reply:
Hey @davids5
What name do you refer to?
The `Auto-monitor` is an option that is under the Watchdog Timer Support option.
The `Auto-monitor` provides 4 mechanisms to feed the dog automatically.
I think the `auto` prefix stands for automatically,i.e., without the developer configuring it in execution code and the `monitor` word means that it will monitor your code because if one task gets stuck the timer will expire and the chip will be reseted.

Reply:
Ah, I get it:  It is really  built in monitoring the system and resetting the watchdog.  "Enable Built in System Monitoring to reset the watchdog."

Is the timer one a HW timer off an ISR?  Because WD kicking done off an ISR is a really bad idea.

Reply:
"Enable Built in System Monitoring to reset the watchdog." is a better title indeed.

The timer option calls `wd_start` in the beginning. 

> This function adds a watchdog timer to the active timer queue.

 and registers a function that will be called from the interrupt level after the specified number of ticks has elapsed. 
When this option is selected, the registered function resets the watchdog and calls `wd_start` again registering itself again.

It is implemented in `drivers/timers/watchdog.c`

Reply:
> "Enable Built in System Monitoring to reset the watchdog." is a better title indeed.
I am easily confused but what I read a lot so I am glad it makes sense, 

> registers a function that will be called from the interrupt level after the specified number of ticks has elapsed.

Yikes, I would put a caution all over that to not to not use it, less one really knows what they are doing.,,,,,